element_lab/export/nwb.py

```diff
@@ -0,0 +1,102 @@
+from element_lab import lab
+
+
+def lab_to_nwb_dict(lab_key=None):
+    """
+    Generate a dictionary containing all relevant lab and institution info
+    :param lab_key: Key specifying one entry in element_lab.lab.Lab
+    :return: dictionary with NWB parameters
+    """
+    if lab_key is not None:
+        lab_info = (lab.Lab & lab_key).fetch1()
+        return dict(
+            institution=lab_info.get('institution', ''),
+            lab=lab_info.get('lab_name', '')
+        )
+    else:
+        return {}
+
+
+def project_to_nwb_dict(project_key=None):
+    """
+    Generate a dictionary object containing relevant project information
+        (e.g., experimental description, related publications, etc.).
+    :param project_key: Key specifying one entry in element_lab.lab.Project
+    :return: dictionary with NWB parameters
+    """
+    if project_key is not None:
+        proj_info = (lab.Project & project_key).fetch1()
+        proj_keyw = (lab.Project.Keywords() & project_key
+                     ).fetch('keyword').tolist()
+        proj_pubs = (lab.Project.Publication() & project_key
+                     ).fetch('publication').tolist()
+        return dict(
+            experiment_description=(proj_info.get('project_description', '')),
+            keywords=proj_keyw,
+            related_publications=proj_pubs
+        )
+    else:
+        return {}
+
+
+def protocol_to_nwb_dict(protocol_key=None):
```

Comment:
This function returns `{}` by default. Perhaps it should not be called without the protocol key.

Reply:
That's a great point! I've updated each of these functions to only execute if the relevant input is provided to the main `elementlab_nwb_dict` function.